lsp-mode.el

```diff
@@ -1290,7 +1290,7 @@ On other systems, returns path without change."
          (type (url-type url))
          (encoding (or locale-coding-system 'utf-8))
          (file (decode-coding-string (url-filename url) encoding))
-         (file-name (if (and type (not (string= type "file")))
```

Comment:
why is this? string= should be more performant in this case?

Reply:
When I try benchmark code with
``` elisp
(setq a '(("aaa" . "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
          ("aaa" . "aab")
          (#("aaa" 0 1 (s a)) . "aaa")
          (#("aaa" 0 1 (s a)) . "aab")))
(benchmark 1000000 '(--map (equal (car it) (cdr it)) a))
;; "Elapsed time: 1.544581s"
(benchmark 1000000 '(--map (string= (car it) (cdr it)) a))
;; "Elapsed time: 1.593952s"
```
It seems `equal` is a little bit faster?